tests/unit/pyramid_sanity/__init___test.py

```diff
@@ -1,114 +1,60 @@
-from unittest.mock import patch
+from unittest.mock import call, patch
 
 import pytest
-from pyramid.config import PHASE3_CONFIG, Configurator
-from pyramid.interfaces import ITweens
+from pyramid.config import Configurator
+from pyramid.tweens import MAIN
 
-from pyramid_sanity import (
-    EgressTweenFactory,
-    IngressTweenFactory,
-    _add_tween,
-    includeme,
-)
+from pyramid_sanity import includeme
 
 
 class TestIncludeMe:
-    def test_it_registers_the_config_object(self, pyramid_config, SanitySettings):
+    def test_it_puts_the_settings_into_the_registry(
+        self, pyramid_config, SanitySettings, settings
+    ):
         includeme(config=pyramid_config)
 
         SanitySettings.from_pyramid_settings.assert_called_once_with(
             pyramid_config.registry.settings
         )
 
-        assert (
-            pyramid_config.registry.settings["pyramid_sanity"]
-            == SanitySettings.from_pyramid_settings.return_value
-        )
+        assert pyramid_config.registry.settings["pyramid_sanity"] == settings
+
+    @pytest.mark.parametrize(
+        "ingress_required,egress_required",
+        [(True, True), (False, True), (True, False), (False, False),],
+    )
+    def test_it_adds_the_tweens(
+        self, ingress_required, egress_required, pyramid_config, settings
+    ):
+        settings.ingress_required = ingress_required
+        settings.egress_required = egress_required
 
-    def test_it_registers_the_tween_factory(self, pyramid_config):
-        # This test is kind of nonsense, just testing the code does what it
-        # does.
         includeme(config=pyramid_config)
 
-        pyramid_config.action.assert_called_once_with(
-            ("tween", "pyramid_sanity.tween_factory", True),
-            _add_tween,
-            args=(pyramid_config,),
-            order=PHASE3_CONFIG,
+        ingress_tween_added = (
+            call("pyramid_sanity.IngressTweenFactory")
+            in pyramid_config.add_tween.call_args_list
         )
+        assert ingress_tween_added == ingress_required
 
-    @pytest.fixture
-    def SanitySettings(self, patch):
-        return patch("pyramid_sanity.SanitySettings")
+        egress_tween_added = (
+            call("pyramid_sanity.EgressTweenFactory", over=MAIN)
+            in pyramid_config.add_tween.call_args_list
+        )
+        assert egress_tween_added == egress_required
```

Comment:
I looked into using a real configurator for this test but couldn't see a way to do the asserts. So just went with mocking it